packages/react-native-codegen/src/parsers/typescript/modules/index.js

```diff
@@ -257,6 +257,17 @@ function translateTypeAnnotation(
       }
     }
     case 'TSTypeLiteral': {
+      // if there is TSIndexSignature, then it is a dictionary
+      if (
+        typeAnnotation.members &&
+        typeAnnotation.members.filter(
+          member => member.type === 'TSIndexSignature',
+        ).length > 0
+      ) {
+        // no need to do further checking
+        return emitObject(nullable);
```

Comment:
from @christophpurrer:
> What if we emit a new type here?
Instead of `GenericObjectTypeAnnotation` we can use `IndexedObjectTypeAnnotation`

Reply:
I think this is the idea of `DictionaryObjectTypeAnnotation` but you said you don't want me to add that type am I right?

Reply:
I don't have a strong opinion of adding or not adding a new type for indexed objects at this point.

If we would add a new type we should keep its name as close as possible to the internal representation in flow-parser (ObjectTypeIndexer) or babel-parser with the TypeScript plugin (TSIndexSignature)

Dictionary is not a term used in either of the 2 parsers.